test/e2e/managementstate_test.go

```diff
@@ -141,11 +141,27 @@ func TestRemovedToManagedTransition(t *testing.T) {
 	}
 
 	t.Log("updating ManagementState to Managed with no storage config")
-	cr.Spec.ManagementState = operatorapi.Managed
-	cr.Spec.Storage = imageregistryv1.ImageRegistryConfigStorage{}
-	if _, err = te.Client().Configs().Update(
-		context.Background(), cr, metav1.UpdateOptions{},
-	); err != nil {
+	err = wait.Poll(time.Second, framework.AsyncOperationTimeout, func() (stop bool, err error) {
```

Comment:
It might be better to use `retry.RetryOnConflict`

Reply:
Way better. Done.